src/main/java/org/mitre/synthea/world/concepts/Terminology.java

```diff
@@ -0,0 +1,575 @@
+      for (int i = 0; i < concepts.getLength(); i++) {
+        Node currentNode = concepts.item(i);
+        String system = currentNode.getAttributes().getNamedItem("codeSystem").getTextContent();
+        String code = currentNode.getAttributes().getNamedItem("code").getTextContent();
+        String display = currentNode.getAttributes().getNamedItem("displayName").getTextContent();
+
+        // the toggle checks to see if we already have the system present
+        // in the ValueSet so we don't have duplicates with the codes
+        // spread out between them.
+        boolean toggle = false;
+
+        for (ValueSet.ConceptSetComponent c : composeComponent.getInclude()) {
+          if (c.getSystem().equals(system)) {
+            c.addConcept().setCode(code).setDisplay(display);
+            toggle = true;
+          }
+        }
+        if (!toggle) {
+          composeComponent.addInclude()
+              .setSystem(system)
+              .addConcept()
+              .setCode(code)
+              .setDisplay(display);
+        }
+      }
+      vs.setCompose(composeComponent);
+
+      // Writes the value set to the valueset folder with VSAC appended.
+      FileWriter w = new FileWriter(
+          Paths.get(Objects.requireNonNull(ClassLoader.getSystemClassLoader()
+              .getResource(VS_FOLDER)).toURI()) + "/" + displayName + "VSAC.json");
+
+      // System.out.println(ctx.newJsonParser().setPrettyPrint(true).encodeResourceToString(vs));
+      ctx.newJsonParser().setPrettyPrint(true).encodeResourceToWriter(vs,w);
+      return vs;
+
+    }
+
+    <E> Optional<E> getRandom(Collection<E> e) {
+      return e.stream()
+                .skip((int) (e.size() * Math.random()))
+                .findFirst();
+    }
+
+    Code chooseCode(Multimap<String, Code> codes) {
+      // Chooses a random code by picking a system then picking a code from
+      // that system.
+
+      List<String> keys = new ArrayList<>(codes.keySet());
+      Random r = new Random();
+      String system = keys.get(r.nextInt(keys.size()));
+      return getRandom(codes.get(system)).get();
+    }
+
+
+
+    String getAuthToken() {
+      return this.authToken;
+    }
+
+    OkHttpClient getSessionClient() {
+      return this.client;
+    }
+
+    void setAuthToken(String token) {
+      this.authToken = token;
+    }
+  }
+
+  static Map<String, String> loadLookupTable() {
+
+    // Loads valueSets from file and puts them in a map of {url : ValueSet}
+    Map<String, String> retVal = new ConcurrentHashMap<>();
+
+    URL valueSetsFolder = ClassLoader.getSystemClassLoader().getResource("code_system_lookup.json");
+    try {
+      JsonObject codeSystemLookup = loadJsonFile(Paths.get(valueSetsFolder.toURI()));
+      for(String member : codeSystemLookup.keySet()) {
+        retVal.put(member,codeSystemLookup.get(member).getAsString());
+      }
+    } catch (URISyntaxException e) {
+      e.printStackTrace();
+    }
+    return null;
```

Comment:
I put this function on the backburner because I wasn't sure if swapping OID for URL was really something that was necessary or if doing it with a lookup table was necessarily the best way.